### PR TITLE
Fix bottom drag bug

### LINKED ISSUE #217: Cannot specify bottom drag in `ocean_simulation`
MWE:

```julia
julia> grid = LatitudeLongitudeGrid(size = (10, 10, 10), latitude = (-10, 10), longitude = (-10, 10), z = (-10, 0), halo = (6, 6, 2))
10×10×10 LatitudeLongitudeGrid{Float64, Bounded, Bounded, Bounded} on CPU with 6×6×2 halo and with precomputed metrics
├── longitude: Bounded  λ ∈ [-10.0, 10.0] regularly spaced with Δλ=2.0
├── latitude:  Bounded  φ ∈ [-10.0, 10.0] regularly spaced with Δφ=2.0
└── z:         Bounded  z ∈ [-10.0, 0.0]  regularly spaced with Δz=1.0

julia> sim = ocean_simulation(grid; bottom_drag_coefficient = 0.001)
ERROR: MethodError: no method matching default_or_override(::Float64)

Closest candidates are:
  default_or_override(::ClimaOcean.OceanSimulations.Default, ::Any)
   @ ClimaOcean ~/development/ClimaOcean.jl/src/OceanSimulations/OceanSimulations.jl:27
  default_or_override(::ClimaOcean.OceanSimulations.Default)
   @ ClimaOcean ~/development/ClimaOcean.jl/src/OceanSimulations/OceanSimulations.jl:27

Stacktrace:
 [1] ocean_simulation(grid::LatitudeLongitudeGrid{…}; Δt::Float64, closure::Oceananigans.TurbulenceClosures.TKEBasedVerticalDiffusivities.CATKEVerticalDiffusivity{…}, free_surface::SplitExplicitFreeSurface{…}, reference_density::Int64, rotation_rate::Float64, gravitational_acceleration::Float64, bottom_drag_coefficient::Float64, forcing::@NamedTuple{}, coriolis::HydrostaticSphericalCoriolis{…}, momentum_advection::VectorInvariant{…}, tracer_advection::FluxFormAdvection{…}, verbose::Bool)
   @ ClimaOcean.OceanSimulations ~/development/ClimaOcean.jl/src/OceanSimulations/OceanSimulations.jl:89
 [2] top-level scope
   @ REPL[3]:1
Some type information was truncated. Use `show(err)` to see complete types.

julia>
```

### PR DESCRIPTION
**Pull request by simone-silvestri**
closes #217

### CODE REVIEW

File: src/OceanSimulations/OceanSimulations.jl [hunk 1]
```diff
@@ -25,6 +25,7 @@ end
 
 Default() = Default(nothing)
 default_or_override(default::Default, value=default.value) = value
+default_or_override(user_specified_value::Number, args...) = user_specified_value
```
Comment (glwagner) [inline src/OceanSimulations/OceanSimulations.jl:28]: Delete this because this is not the design

Comment (glwagner) [inline src/OceanSimulations/OceanSimulations.jl:28]: This isn't quite right. Can I commit to this PR? I have an idea to make the design better.

Comment (simone-silvestri) [reply src/OceanSimulations/OceanSimulations.jl:28]: for sure

**Comment by glwagner** [comment]
@simone-silvestri this PR does not add a test. I think we should get in the habit of adding a test when we find a bug to ensure the bug doesn't crop up again. Hopefully that will help us start to move a bit faster...

**Comment by glwagner** [comment]
I added this docstring:

```julia
"""
    default_or_override(default::Default, possibly_alternative_default=default.value) = possibly_alternative_default
    default_or_override(override, alternative_default) = override

Either return `default.value`, or an override.

The override may be specified two ways:

1. Provide `value` by writing `default_or_override(default, override_value)`. This
pattern is useful for constructors, where "alternative defaults" must be implemented based
on other aspects of a simulation's configuration.

2. Provide the override in place of `default` by writing either `default_or_override(override)` or
`default_or_override(override, alternative_default)`. In either of the above two cases, provided that
`!(override isa Default)`, then `override` will supercede the `alternative_default`.
"""
default_or_override(default::Default, possibly_alternative_default=default.value) =  possibly_alternative_default
default_or_override(override, alternative_default=nothing) = override
```

**Comment by glwagner** [comment]
I'll merge this, then add some tests for `ocean_simulation` in a new PR. We need to test every kwarg.

**Comment by navidcy** [comment]
@glwagner merge?

**Review by glwagner — APPROVED** [review]

**Pull request merged**